lab_1/main.py

```diff
@@ -4,125 +4,114 @@
+        check_l = not isinstance(lcs, bool) and isinstance(lcs, int) and lcs > 0
+        rcs = right_context_size
+        check_r = not  isinstance(rcs, bool) and isinstance(rcs, int) and rcs > 0
+        idx = [i for i, x in enumerate(tokens) if x == word]
+        if check_l and check_r:
+            conc = [tokens[i-lcs:i+rcs+1] for i in idx]
+        elif not check_l and check_r:
+            conc = [tokens[i:i+rcs+1] for i in idx]
+        elif check_l and not check_r:
+            conc = [tokens[i-lcs:i+1] for i in idx]
+    return conc
+
+
+def get_adjacent_words(tokens: list,
+                       word: str,
+                       left_n: int,
+                       right_n: int) -> list:
+    conc = get_concordance(tokens, word, left_n, right_n)
+    if not left_n:
+        output = [[elem[-1]] for elem in conc]
+    elif not right_n:
+        output = [[elem[0]] for elem in conc]
+    else:
+        output = [[elem[0], elem[-1]] for elem in conc]
+    return output
 
 
 def read_from_file(path_to_file: str) -> str:
-    """
-    Opens the file and reads its content
-    :return: the initial text in string format
-    """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
-
+    with open(path_to_file, 'r', encoding='utf-8') as file:
+        data = file.read()
     return data
 
 
 def write_to_file(path_to_file: str, content: list):
-    """
-    Writes the result in a file
-    """
-    pass
-
-
-def sort_concordance(tokens: list, word: str, left_context_size: int, right_context_size: int, left_sort: bool) -> list:
-    """
-    Gets a concordance of a word and sorts it by either left or right context
-    :param tokens: a list of tokens
-    :param word: a word-base for a concordance
-    :param left_context_size: the number of words in the left context
-    :param right_context_size: the number of words in the right context
-    :param left_sort: if True, sort by the left context, False – by the right context
-    :return: a concordance
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
-                    'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad']
-    word = 'happy'
-    left_context_size = 2
-    right_context_size = 3
-    left_sort = True
-    --> [['dog', 'is', 'happy', 'but', 'the', 'cat'], ['man', 'is', 'happy', 'the', 'dog', 'is']]
-    """
-    pass
+    with open(os.path.join(path_to_file, 'report.txt'),
+              'w', encoding='utf-8') as file:
+        file.write('\n'.join([' '.join(k) for k in content]))
+
+
+def sort_concordance(tokens: list,
+                     word: str,
+                     left_context_size: int,
+                     right_context_size: int,
+                     left_sort: bool) -> list:
+    output = []
+    if isinstance(left_sort, bool):
+        lcs = left_context_size
+        rcs = right_context_size
+        conc = get_concordance(tokens, word, lcs, rcs)
+        try:
```

Comment:
try-except solution is applicable for one-line code where we expect errors, not whole main logic implementation. Consider writing conditions or other checks.

Reply:
fixed